src/dimred/lsi/test.py

```diff
@@ -0,0 +1,210 @@
+import sys
+import pytest
+import subprocess
+import mudata as mu
+import numpy as np
+
+## VIASH START
+meta = {
+    'resources_dir': 'resources_test',
+    'executable': './target/docker/dimred/lsi/lsi',
+    'config': './src/dimred/lsi/config.vsh.yaml'
+}
+## VIASH END
+
+input_path = f"{meta['resources_dir']}/concat_test_data/e18_mouse_brain_fresh_5k_filtered_feature_bc_matrix_subset_unique_obs.h5mu"
+
+
+'''
+Tests: 
+1. general test 
+2. test HVF
+3. test modality
+4. test layer
+5. test overwrite 
+'''
+
+@pytest.fixture
+def atac_mudata(tmp_path):
+    
+    mdata = mu.read_h5mu(input_path)
+    mdata.mod["atac"].layers["counts"] = mdata.mod["atac"].X
+    mdata.mod["atac"].var["highly_variable"] = np.random.choice([True, False], size=mdata.mod["atac"].n_vars)
+    print(mdata)
+
+    mdata.write(tmp_path / "atac_mudata.h5mu")
+
+    return tmp_path / "atac_mudata.h5mu"
+
+# 1.general test
+def test_lsi(tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", input_path,
+     "--output", str(output_path),
+     "--obsm_output", "X_test",
+     "--num_components", "30"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_test" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_test"].shape == (data.mod["atac"].n_obs, 30)
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+
+
+
+# 2.test HVF 
+def test_select_highly_variable_column(atac_mudata, tmp_path):
+    output_path = tmp_path / "output_lsi.h5mu"
+
+    # run component
+    cmd_args = [
+    meta["executable"],
+     "--input", str(atac_mudata),
+     "--output", str(output_path),
+     "--var_input", "highly_variable"
+    ]
+    subprocess.run(cmd_args, check=True)
+    
+    assert output_path.is_file()
+    data = mu.read_h5mu(output_path)
+    assert "X_lsi" in data.mod['atac'].obsm
+    assert data.mod["atac"].obsm["X_lsi"].shape == (data.mod["atac"].n_obs, 50)
+    assert "highly_variable" in data.mod["atac"].var.columns
+    assert "lsi" in data.mod['atac'].uns
+    assert "LSI" in data.mod['atac'].varm
+    assert data.mod["atac"].varm["LSI"].shape == (data.mod["atac"].n_vars, 50)
+
+
+def test_highly_variable_column_does_not_exist_raises():
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        cmd_args = [
+        meta["executable"],
+        "--input", input_path,
+        "--output", "output_lsi.h5mu",
+        "--var_input", "does_not_exist"
+        ]
+        subprocess.run(cmd_args, check=True)
+        
+        assert "ValueError: Requested to use .var column 'does_not_exist' as a selection of genes, but the column is not available." in \
+            err.value.stdout.decode('utf-8')
```

Comment:
Be careful with indentation when using `pytest.raises`! This assert is being skipped because `subprocess.run` raises `CalledProcessError` and the rest of the code block is skipped. pytest.raises` catches the `CalledProcessError`. So this means that this test will always succeed!
```suggestion
      assert "ValueError: Requested to use .var column 'does_not_exist' as a selection of genes, but the column is not available." in \
          err.value.stdout.decode('utf-8')
```